### PR TITLE
[Issue-1059] - adds ui/api for retrieving cache stats (edges and mids). i.e. health,…

### LINKED ISSUE #1059: [TC-421] TPv2 - create cache health view
<p>In TPv2, create a view similar to the edge health (rascal status) view, however, include mid caches as well - <a href="http://localhost:3000/edge_health" class="external-link" rel="nofollow">http://localhost:3000/edge_health</a> - of the current TO UI.</p>

<p>This will probably require API work in addition to UI work.</p>

Author: Jeremy Mitchell
JIRA Link: <a href="https://issues.apache.org/jira/browse/TC-421">https://issues.apache.org/jira/browse/TC-421</a>

### PR DESCRIPTION
**Pull request by mitchell852**
… connections and bandwidth

fixes #1059

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/405/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/361/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/410/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/366/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/411/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/367/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/413/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/414/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/369/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/370/

File: traffic_ops/app/lib/API/Cache.pm [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+	}
+	else {
+		return 0;
+	}
+}
+
+sub int_or_zero {
+	my $val = shift;
+
+	if ( defined($val) && ( $val =~ /^\d+?$/ || $val =~ /^\d+\.\d+$/ ) ) {
```
Comment (dangogh) [inline traffic_ops/app/lib/API/Cache.pm]: positive ints only?

File: traffic_ops/app/lib/API/Cache.pm [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+			push( @{ $big_obj->{'cachegroups'}->{ $i->{'cachegroup'} }->{'caches'} }, $i->{'host_name'} );
+		}
+	}
+
+	return ($big_obj);
+}
+
+sub def_or_zero {
+	my $val = shift;
+
```
[the remark this answers is not in the document]
Comment (mitchell852) [reply traffic_ops/app/lib/API/Cache.pm]: ok, thanks dan

**Pull request merged**